x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/RemoteClusterPermissionGroup.java

```diff
@@ -107,15 +106,17 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         RemoteClusterPermissionGroup that = (RemoteClusterPermissionGroup) o;
-        return Arrays.equals(clusterPrivileges, that.clusterPrivileges)
-            && Arrays.equals(remoteClusterAliases, that.remoteClusterAliases)
-            && Objects.equals(remoteClusterAliasMatcher, that.remoteClusterAliasMatcher);
+        // The remoteClusterAliasMatcher is intentionally omitted from equals and hashCode implementation.
```

Comment:
nit: thanks for the comment. however we probably don't need a full explanation, just a mention that it is omitted intentionally.  I know it is really pedantic but long comments like this draw your attention as something important to read where all we really need to know is that we intentionally omitted those fields to avoid appearing as a bug and/or accidentally adding those fields here at some point.